### DISCUSSION TITLE
Colors set in preset is not taking priority

### DISCUSSION THREAD

**Discussion by yellow1912** [tailwindlabs/tailwindcss#7948] [Help]
**What version of Tailwind CSS are you using?**

3.0.23

**What build tool (or framework if it abstracts the build tool) are you using?**

postcss 8.4.7

**What version of Node.js are you using?**

14.17.0

**What browser are you using?**

N/A

**What operating system are you using?**

Ubuntu

**Reproduction URL**

I have 2 files (due to my setup), 1 main config file and 1 local config file:
Main file:
https://gist.github.com/yellow1912/09b586feaf695f4e21cb302a8b731ca4

Local file:
https://gist.github.com/yellow1912/c6f128ba151c45856b9246e3cd385894

Scss file:
https://gist.github.com/yellow1912/25fe86245063d8b1bcaa37180d472b4a

**Describe your issue**

When I try to compile tailwind like this:

`npx tailwind -i style.scss -c tailwind.config.js --local-config=local.config.js --load-path=node_modules --content=content.html | npx postcss --o style.min.css
`
I get this error:

> CssSyntaxError: tailwindcss: style.scss:29:20: The `text-test-color-z` class does not exist. If `text-test-color-z` is a custom class, make sure it is defined within a `@layer` directive.

If I remove the color definition in the main config file, then the ones I set in my preset will work. I think either there is a bug or perhaps I misunderstand how this should work?

**Comment by adamwathan** [comment]
Hey! I haven't tried to actually run the files you've shared but there's a lot of weird things going on in your usage here.

- Tailwind can't compile SCSS files, so you shouldn't be passing a `.scss` file to Tailwind
- Piping Tailwind CLI into PostCSS CLI is unnecessary, you should just use one or the other

If you can provide an actual minimal reproduction (a small GitHub project where I can just run one command to see the results you are seeing) it's more likely that someone will be able to find time to help your troubleshoot your problem.

**Comment by yellow1912** [comment]
@adamwathan sorry, it should be css, but as you can see the content is css and it is parsed just fine (which results in the error of the missing color). Removing the pipe does not affect the error result.

I will do some more checking and create a project if necessary.

**Comment by yellow1912** [comment]
I added some hacky code fix to remove the key(s) in the main configs if it is also inside the preset:

```
if (args.hasOwnProperty('local-config')) {
  config.presets = [require(args['local-config'])];

  config.presets.forEach(preset => {
    for (let k in preset.theme) {
      if ('extend' !== k) {
        if (config.theme.hasOwnProperty(k)) {
          delete config.theme[k];
        }
      }
    }
  })
}
```
